ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Model.hs

```diff
@@ -446,18 +444,33 @@ addBlock cfg blk m = Model {
           (Proxy @(BlockProtocol blk))
           (selectView (configBlock cfg) . getHeader)
           (currentChain m)
-      . filter (extendsImmutableChain . fst)
-      $ candidates
+      $ consideredCandidates
 
-addBlocks :: LedgerSupportsProtocol blk
+    -- Sometimes blocks are added not in order and therefore we cannot just look
+    -- at the last added block and push it to the set of known valid blocks. The
+    -- easiest (and considering the lengths of the generated chains) way is to
+    -- just push all the valid blocks in the chain into the set.
+    --
+    -- Also, the chain selection algorithm in the SUT will test chains one by
```

Comment:
I don't know if we should be referring to the SUT behaviour here. I wonder if we could omit the SUT part.